### PR TITLE
Interfaces for offline added to all models (sorry for late PR)

### PR DESCRIPTION
**Pull request by tdybda17**
Related to: https://github.com/aau-giraf/weekplanner/issues/389
**Important (Please read)**

This is the initial work for the offline repository (sorry for the late PR). Please only look at the structure of how to use the repository (The `.get(int id)` in `repository.dart` is implemented and it works - but there are no tests yet!). The reason for this PR is that other groups can get started on their offline-related issues.

The repository can be used on any class implementing the `Model` class in the api_client. The repository current provide a insert, update, delete, get and all (More can be added if needed).

**Example of usage**
To insert a new `Activity` to the offline repository:
```
ActivityModel activity = ActivityModel(...);
ActivityModel.offline().insert(activity);
```

Other considerations:
With this implementation of the offline repository, all design patterns can be used for the repository in the future, if needed.

You will need to mock the methods for all testing.

### CODE REVIEW

File: lib/offline_repository/repository.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    _db = await DbProvider.instance.database;
+    return _db;
+  }
+
+  final String _tableName;
+
+  Database _db;
+
+  @override
+  Future<Model> insert(Model model) {
```
Comment (MrPekar98) [inline lib/offline_repository/repository.dart:28]: Is there a reason the methods insert(), update(), delete(), and all are not implemented?

Comment (tdybda17) [reply lib/offline_repository/repository.dart:28]: Yes https://github.com/aau-giraf/api_client/pull/46#issue-407395781

Comment (MrPekar98) [reply lib/offline_repository/repository.dart:28]: What is the reason then? It's not really clear from the description.

Comment (Simon97) [reply lib/offline_repository/repository.dart:28]: The methods should and will be implemented, but are not required for the other groups with offline issues to begin their work. That's why this PR was made without the methods has been implemented. :)

**Pull request merged**